app/src/main/java/net/aiscope/gdd_app/ui/capture/CaptureImagePresenter.kt

```diff
@@ -6,20 +6,26 @@ import timber.log.Timber
 
 class CaptureImagePresenter(
     val view: CaptureImageView,
-    val repository: SampleRepository
+    val repository: SampleRepository,
+    var receivedImageCapture: Boolean = false
 ) {
 
     fun handleCaptureImageButton(imageName: String) {
-        Timber.tag("Taking Photo").d("button pressed")
-        view.takePhoto(imageName) { file ->
-            Timber.tag("Taking Photo").d(file?.absolutePath)
-            if (file == null) {
-                view.notifyImageCouldNotBeTaken()
-            } else {
-                val sample = repository.current().addNewlyCapturedImage(file)
-                repository.store(sample)
+        if (receivedImageCapture){
+            Timber.tag("Taking Photo").d("Already processing photo")
+        } else {
```

Comment:
You can do a `return` in the `if`, so you don't need the `else` (results in a smaller diff 😛 )

Reply:
Personally not a big fan of using `return` as a break statement but sure